hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/strategy/DayBasedCompactionStrategy.java

```diff
@@ -78,7 +78,8 @@ public List<HoodieCompactionOperation> orderAndFilter(HoodieWriteConfig writeCon
   public List<String> filterPartitionPaths(HoodieWriteConfig writeConfig, List<String> allPartitionPaths) {
     return allPartitionPaths.stream().map(partition -> partition.replace("/", "-"))
         .sorted(Comparator.reverseOrder()).map(partitionPath -> partitionPath.replace("-", "/"))
-        .collect(Collectors.toList()).subList(0, writeConfig.getTargetPartitionsPerDayBasedCompaction());
+        .collect(Collectors.toList()).subList(0, Math.min(allPartitionPaths.size(),
```

Comment:
Good catch! Can you please check the CI failure?

Reply:
Seemingly very unrelated and couldn't reproduce locally. Will trigger again